psydac/linalg/basic.py

```diff
@@ -128,6 +140,11 @@ def domain( self ):
     def codomain( self ):
         pass
 
+    @property
+    @abstractmethod
+    def dtype( self ):
+        pass
+
```

Comment:
See my previous comment.

I am not sure that `LinearOperator` should have the `dtype` method... As far as I can tell only the `Matrix` class strictly needs `dtype`, because it has the `toarray` and `tosparse` methods.

Reply:
we need to know the datatypes of the elements that it accepts, whether it works on integers, complex numbers of floats